--- speech_tools/config/compilers/gcc_defaults.mak.orig	2004-04-21 23:56:35.000000000 +0200
+++ speech_tools/config/compilers/gcc_defaults.mak	2004-04-22 10:44:51.046818928 +0200
@@ -73,7 +73,7 @@
 PROFILE_gprof_CXXFLAGS  = -pg
 PROFILE_gprof_LINKFLAGS = -pg
 
-SHARED_CCFLAGS  = -fpic -fno-shared-data
+SHARED_CCFLAGS  = -fPIC -fno-shared-data
 SHARED_CXXFLAGS  = -fPIC -fno-shared-data
 SHARED_LINKFLAGS = -fno-shared-data
 
