--- speech_tools/speech_class/EST_wave_io.cc	2010-01-03 16:57:14.396879270 +0100
+++ speech_tools/speech_class/EST_wave_io.cc.orig	2006-01-12 16:37:51.000000000 +0100
@@ -68,7 +68,7 @@
 
 int nist_get_param_int(const char *hdr, const char *field, int def_val)
 {
-    char *p;
+    const char *p;
     int val;
 
     if (((p=strstr(hdr,field)) != NULL) &&
@@ -84,7 +84,8 @@
 
 char *nist_get_param_str(const char *hdr, const char *field, const char *def_val)
 {
-    char *p,*val;
+    const char *p;
+    char *val;
     int size;
 
     if (((p=strstr(hdr,field)) != NULL) &&
--- speech_tools/base_class/EST_TSimpleMatrix.cc~	2004-09-30 14:53:35.000000000 +0200
+++ speech_tools/base_class/EST_TSimpleMatrix.cc	2010-01-03 16:48:52.345905810 +0100
@@ -44,6 +44,7 @@
 #include "EST_TVector.h"
 #include <fstream>
 #include <iostream>
+#include <string.h>
 #include "EST_cutils.h"
 
 template<class T> 
--- speech_tools/base_class/EST_TSimpleVector.cc~	2006-07-06 14:57:18.000000000 +0200
+++ speech_tools/base_class/EST_TSimpleVector.cc	2010-01-03 16:51:50.968904972 +0100
@@ -43,6 +43,7 @@
 #include "EST_TSimpleVector.h"
 #include "EST_matrix_support.h"
 #include <fstream>
+#include <string.h>
 #include "EST_cutils.h"
 
 template<class T> void EST_TSimpleVector<T>::copy(const EST_TSimpleVector<T> &a)
