--- speech_tools/configure.in.orig	2002-12-28 12:57:31.000000000 +0100
+++ speech_tools/configure.in	2004-04-18 16:13:01.671787272 +0200
@@ -42,10 +42,14 @@
 
 AC_C_BIGENDIAN
 
-AC_CHECK_LIB(termcap, tputs, [TERMCAPLIB=-ltermcap], [TERMCAPLIB=-lncurses])
-if test "$TERMCAPLIB" != "-ltermcap"; then
-	AC_CHECK_LIB(ncurses, tputs, [TERMCAPLIB=-lncurses], [TERMCAPLIB=-lcurses])
-fi
+TERMCAPLIB=
+AC_CHECK_LIB(tinfo, tputs, [TERMCAPLIB=-ltinfo], [
+  AC_CHECK_LIB(ncurses, tputs, [TERMCAPLIB=-lncurses], [
+    AC_CHECK_LIB(curses, tputs, [TERMCAPLIB=-lcurses], [
+      AC_CHECK_LIB(termcap, tputs, [TERMCAPLIB=-ltermcap])
+    ])
+  ])
+])
 AC_SUBST(TERMCAPLIB)
 
 dnl Which version of GCC do we have here
