--- speech_tools/stats/EST_DProbDist.cc.orig	2004-09-30 14:53:36.000000000 +0200
+++ speech_tools/stats/EST_DProbDist.cc	2010-01-03 17:09:34.651883980 +0100
@@ -310,7 +310,7 @@
     if (type == tprob_discrete)
 	return 0;
     else
-	return (int)scounts.list.head();
+	return scounts.list.index(scounts.list.head());
 }
 
 int EST_DiscreteProbDistribution::item_end(int idx) const
@@ -326,7 +326,7 @@
     if (type == tprob_discrete)
 	return ++idx;
     else
-	return (int)next((EST_Litem *)idx);
+	return scounts.list.index(next((EST_Litem *)idx));
 }
 
 const EST_String &EST_DiscreteProbDistribution::item_name(int idx) const
